sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -656,6 +661,24 @@
       return toBuilder().setCommitOffsetsInFinalizeEnabled(true).build();
     }
 
+    /**
+     * Set additional configuration for the backend offset consumer. It may be required for a
+     * secured Kafka cluster, especially when you see similar WARN log message 'exception while
+     * fetching latest offset for partition {}. will be retried'.
+     *
+     * <p>In {@link KafkaIO#read()}, there're two consumers running in the backend actually:<br>
+     * 1. the main consumer, which reads data from kafka;<br>
+     * 2. the secondary offset consumer, which is used to estimate backlog, by fetching latest
+     * offset;<br>
+     *
+     * <p>By default, offset consumer inherits the configuration from main consumer, with an
+     * auto-generated {@link ConsumerConfig#GROUP_ID_CONFIG}. This may not work in a secured Kafka
+     * which requires more configurations.
+     */
```

Comment:
Also, we could let AUTO_COMMIT overridden by the user. It is not related to offsetConsumer functionality. I had disabled it just not to confuse the user with this internal consumer with randomized name.

Reply:
Ahh, ok. So, it should be moved up, before `if (source.getSpec().getOffsetConsumerConfig() != null)` condition.

Reply:
ok for me to move forward `AUTO_COMMIT`. But is it necessary to explain `ISOLATION_LEVEL_CONFIG`?  It's too much about backend code to me.

Reply:
I agree. I don't think we need to mention about ISOLATION_LEVEL_CONFIG in JavaDoc.

Reply:
I was worried about that because, imo, it's kind of hidden and not evident behaviour. Anyway, it was just recommendation, so I'm ok to keep javadoc as it is.